src/components/find-on-shelf/FindOnShelfPeriodicalDropdowns.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { FC, useState } from "react";
+import ExpandMoreIcon from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/ExpandMore.svg";
+import { HoldingsForBibliographicalRecordV3 } from "../../core/fbs/model";
+import makePeriodicalEditionsFromHoldings from "../material/helper";
+import { groupObjectArrayByProperty } from "../../core/utils/helpers/general";
+
+export interface FindOnShelfPeriodicalDropdownProps {
+  manifestationsHoldings: HoldingsForBibliographicalRecordV3[];
+  setSelectedPeriodical: (selectedWeek: string | null) => void;
+}
+
+const FindOnShelfPeriodicalDropdown: FC<FindOnShelfPeriodicalDropdownProps> = ({
+  manifestationsHoldings,
+  setSelectedPeriodical
+}) => {
+  const periodicalEditionsArray = makePeriodicalEditionsFromHoldings(
+    manifestationsHoldings[0].holdings
+  );
+  const sortedperiodicalEditionsArray = groupObjectArrayByProperty(
+    periodicalEditionsArray,
+    "volumeYear"
+  );
+
+  const [selectedYear, setSelectedYear] = useState<string>(
+    String(Object.keys(sortedperiodicalEditionsArray).sort().pop())
+  );
+
+  return (
+    <div className="modal-find-on-shelf__periodical-dropdowns">
+      <div className="dropdown dropdown--grey-borders">
+        <select
+          id="find-on-shelf-periodical-year"
+          className="dropdown__select"
+          aria-label="Choose periodical year"
+          defaultValue={selectedYear}
+          onChange={(e) => setSelectedYear(e.target.value)}
+        >
+          {Object.keys(sortedperiodicalEditionsArray)
+            .sort()
+            .map((item) => (
```

Comment:
Consider how you can improve the readability of this:

1. We `sort()` something that based on the variable name `sortedperiodicalEditionsArray` already seems to be sorted. You might say that it has not been sorted - it has been grouped.
2. The name `item` says nothing about what the value contains. It is in fact a (volume) year.
3. `Array` can become a superfluous or misleading suffix. Giving the variable a plural name sortedperiodicalEdition*s* indicate that we have many. This will usually be in the from of an array or set. Also in this case the value is in fact not an array. It is an object used as a [map](https://javascript.info/map-set).

One approach would be to move most of this out of the JSX and colocate it with the grouping.

Reply:
Very very good comment. Thanks @kasperg :)